py3status/modules/deadbeef.py

```diff
@@ -67,56 +79,48 @@ def _is_running(self):
 
     def deadbeef(self):
         color = self.color_stopped
-        artist = isplaying = length = playback_time = title = \
-            tracknumber = year = ""
+        status = ''
 
         if not self._is_running():
             return {
                 'cached_until': self.py3.time_in(self.cache_timeout),
-                'color': color,
-                'full_text': ''
+                'full_text': self.py3.safe_format(self.format),
+                'color': color
             }
 
-        # mix metadata fields in with the delimiters
-        fmt = self.delimiter.join([
-            '%artist%', '%isplaying%', '%length%', '%playback_time%',
-            '%title%', '%tracknumber%', '%year%'
-        ])
+        # merge format placeholders with format separators
+        # merge hidden placeholders with control separators
+        # merge strings together to run deadbeef command only once
+        out = FORMAT_SEPARATOR.join(['%{}%'.format(x) for x in self.placeholders])
+        control = [CONTROL_SEPARATOR + s for s in (CONTROL_PARAMETERS)]
+        fmt = ''.join(out) + ''.join(control)
```

Comment:
you are just making extra work and complication here.  title and is playing are not special in any way other than you always want them.  Simple is better